src/flamenco/runtime/context/fd_exec_txn_ctx.h

```diff
@@ -39,6 +39,32 @@ typedef struct fd_vote_account_cache_entry fd_vote_account_cache_entry_t;
 #define MAP_KEY_HASH(key,seed) ( ((key)->ui[0]) ^ (seed) )
 #include "../../../util/tmpl/fd_map_chain.c"
 
+/* An entry in the instruction trace */
+struct fd_exec_instr_trace_entry {
+  /* Metadata about the instruction */
+  fd_instr_info_t * instr_info;
+  /* Stack height when this instruction was pushed onto the stack (including itself)
+     https://github.com/anza-xyz/agave/blob/d87e23d8d91c32d5f2be2bb3557c730bee1e9434/sdk/src/transaction_context.rs#L475-L480 */
+  ulong stack_height;
+};
+typedef struct fd_exec_instr_trace_entry fd_exec_instr_trace_entry_t;
+
+#define FD_EXEC_INSTR_TRACE_ENTRY_ALIGN (8UL)
+#define FD_EXEC_INSTR_TRACE_ENTRY_SIZE  (16UL)
+
+struct fd_instr_info_pool_elem {
+  ulong next;
+  fd_instr_info_t info;
+};
+typedef struct fd_instr_info_pool_elem fd_instr_info_pool_elem_t;
```

Comment:
I was torn on this.. I went with pool in the end because it encapsulates the pointer handling, and gives us flexibility in the future to change our usage patterns, but you're right, it was probably the wrong choice.

The main downside to using a pool is performance when iterating over the traces, (and the space for the pointers), but since this is a rare operation I thought this was acceptable.

Either way, it's easy to change - let's see how the RPC/logging stuff evolves as that will use a similar data structure and may end up re-using some of this.